featuretools/primitives/standard/transform/time_series/box_cox.py

```diff
@@ -0,0 +1,51 @@
+from scipy import stats
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Double
+
+from featuretools.primitives.base import TransformPrimitive
+
+
+class BoxCox(TransformPrimitive):
+    """Applies a transformation to help your data resemble a normal distribution,
+        commonly known as the Box Cox transformation. The input data must be positive.
+        It cannot contain zeroes or negative numbers.
+        Commonly used with time series data, to reduce the variance.
+
+    Args:
+        lmbda (int, float, optional): The exponent to use for the transformation.
+            Default is None.
+
+    Examples:
+        >>> box_cox = BoxCox()
+        >>> transformed = box_cox([1, 10, 5, 4, 2])
+        >>> transformed = [round(x, 2) for x in transformed.tolist()]
+        >>> transformed
+        [0.0, 2.69, 1.79, 1.52, 0.73]
+
+        You can specify the lambda to use for the tranformation
+
+        >>> box_cox = BoxCox(lmbda=-1)
+        >>> transformed = box_cox([1, 10, 5, 4, 2])
+        >>> transformed = transformed.tolist()
+        >>> transformed
+        [0.0, 0.9, 0.8, 0.75, 0.5]
+
+    """
+
+    name = "box_cox"
+    input_types = [
+        ColumnSchema(semantic_tags={"numeric"}),
+    ]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
+    uses_full_dataframe = True
+
+    def __init__(self, lmbda=None):
+        self.lmbda = lmbda
+
+    def get_function(self):
+        def box_cox(numeric):
+            if self.lmbda is not None:
+                return stats.boxcox(numeric, lmbda=self.lmbda)
+            return stats.boxcox(numeric)[0]
```

Comment:
I'm not exactly sure how best to go about this, but how does this work in a train/test split scenario where a user might want to use this primitive on the training set and then separately later on a test set? I don't think there is any way for them to access the lambda value that was automatically determined so they could serialize the primitive and apply it to new data and have the new data transformed in the same way.

Also, if this primitive is applied to multiple numeric columns each column could end up with a unique lambda value, which presents additional challenges when knowing what lambda value actually got applied to each column.

I'm on the fence about whether we should automatically determine lambda or whether we should require the user to provide that value for now until we can sort this out better.